### PR TITLE
Integrate less.js v1.6.3

### LINKED ISSUE #24: Running build twice produces vastly different CSS output with less-openui 0.5.2
OpenUI5 version: 1.54.4 (with less-openui5 0.5.2)

Browser/version (+device/version): N/A

Any other tested browsers/devices(OK/FAIL): N/A

URL (minimal example if possible):  N/A

User/password (if required and possible - do not post any confidential information here):  N/A

Steps to reproduce the problem: 
1.  Run build of an OpenUI5 library project by calling grunt build
2.  Rename the target directory target.one
3.  Run the build again
4.  Compare the target and target.one directories

What is the expected result?  Contents of the directories from the two different builds are the same.

What happens instead?  Contents can be vastly different.

Any other information? (attach screenshot if possible)
In our case, the project being built has a reasonably complex .less inheritance/import chain.  There could be up to 5 levels of inheritance to a single theme.  For example, the inheritance chain for one of our themes (from the less files with least precedence to most) is:
base > sap_belize > sas_base > sas_corporate

It seems like less files are getting loaded or processed in different orders between builds.  We don’t know if this has to do with the asynchronous file loading introduced between 0.2.0 and 0.5.2.  In our debugging, we put log messages in lib/fileUtils – before line 64, which is the fs.readFile(fileInfo.path, { line and before line 70, which is the resolve({ line and saw the following for base.less:

Debug statements when incorrect CSS generated:
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less 
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less 
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less 
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less

Debug statements when correct CSS generated:
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less 
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less 
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less 
Reading file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\base\base.less 
Resolving file: C:\dev\master\htmlcommons-themes\openui-framework-themes\build\tmp\build\unpackedOpenuiCompileAndProvidedDependencies\resources\sap\ui\core\themes\sap_belize\base.less

When sap_belize is “resolved” last, the resulting CSS is as expected.  When base, is “resolved” last, the CSS is incorrect.  We haven’t investigated what happens in “resolve”, but it looks like the last base.less “resolved” is the one that takes precedence in the CSS output.

### LINKED ISSUE #25: NPM warn security issue
Hi,

When using less-openui5 dependency, a security issue raised:
_**npm WARN notice [SECURITY] mime has 1 moderate vulnerability. Go here for more details: https://nodesecurity.io/advisories?search=mime&version=1.2.11 - Run `npm i npm@latest -g` to upgrade your npm version, and then `npm audit` to get more info.**_

To fix this issue the dependency of less should be updated to 3.x versions,
because less@3.0.0 uses mime@^1.4.1

Thanks,
Gaia

### PR DESCRIPTION
**Pull request by matz3**
It is not possible to upgrade less to a higher version for compatibility reasons.
Therefore less 1.6.3 sources were included with the relevant fix for a race condition bug.
Fixes #24 

This allows us to update/remove outdated optional dependencies of the less library.
Fixes #25 

The respective options 'sourceMap' and 'cleancss' which required 2 of these dependencies now result in an error when provided.

### CODE REVIEW

File: lib/less/fileLoader.js [hunk 1]
```diff
@@ -6,8 +6,9 @@
  * Licensed under the Apache v2 License.
  *
  *
- * This file contains a modified version of the function "less.Parser.fileLoader"
- * which has been taken from LESS v1.6.3 (https://github.com/less/less.js/blob/v1.6.3/lib/less/index.js#L127-L233).
+ * This file contains a modified version of the local function "imports.push" within the constructor of "less.Parser"
+ * which has been taken from LESS v1.6.3 (https://github.com/less/less.js/blob/v1.6.3/lib/less/parser.js#L70-L111)
+ * and by applying commit https://github.com/less/less.js/commit/ccd8ebbfdfa300b6e748e8d7c12e3dbb0efd8371.
```
Comment (tobiasso85) [inline lib/less/fileLoader.js]: applied

File: lib/less/importsPush.js [hunk 2]
```diff
@@ -7,15 +7,16 @@
  *
  *
  * This file contains a modified version of the local function "imports.push" within the constructor of "less.Parser"
- * which has been taken from LESS v1.6.3 (https://github.com/less/less.js/blob/v1.6.3/lib/less/parser.js#L70-L111).
+ * which has been taken from LESS v1.6.3 (https://github.com/less/less.js/blob/v1.6.3/lib/less/parser.js#L70-L111)
+ * and by applying commit https://github.com/less/less.js/commit/ccd8ebbfdfa300b6e748e8d7c12e3dbb0efd8371.
```
Comment (tobiasso85) [inline lib/less/importsPush.js]: applied

File: lib/less/fileLoader.js (hunk 1 above)
Comment (matz3) [reply lib/less/fileLoader.js]: Done

File: lib/less/importsPush.js (hunk 2 above)
Comment (matz3) [reply lib/less/importsPush.js]: Done

File: README.md [hunk 3]
```diff
@@ -149,6 +149,8 @@ Type `object`
 
 Options for the [less](http://lesscss.org) compiler (`tree.toCss`).
 
+**Note:** The options `sourceMap` and `cleancss` are not supported!
```
Comment (RandomByte) [inline README.md]: Maybe add an info that they will throw an exception when used

File: lib/less/importsPush.js [hunk 4]
```diff
@@ -7,15 +7,16 @@
...
+ * and applied commit https://github.com/less/less.js/commit/ccd8ebbfdfa300b6e748e8d7c12e3dbb0efd8371.
  * Modifications are marked with comments in the code.
  *
  */
 
 /* eslint-disable consistent-this, new-cap, no-invalid-this */
 "use strict";
 
-const less = require("less");
+const less = require("../thirdparty/less");
```
Comment (RandomByte) [inline lib/less/importsPush.js]: This modification should be marked

File: lib/less/fileLoader.js [hunk 5]
```diff
@@ -7,7 +7,8 @@
  *
  *
  * This file contains a modified version of the function "less.Parser.fileLoader"
- * which has been taken from LESS v1.6.3 (https://github.com/less/less.js/blob/v1.6.3/lib/less/index.js#L127-L233).
```
[the remark this answers is not in the document]
Comment (matz3) [reply lib/less/fileLoader.js]: Yes, it's indeed not relevant for this one. So you would just not mention it here?

[the remark this answers is not in the document]
Comment (RandomByte) [reply lib/less/fileLoader.js]: Yeah, the README already mentions it in general. Then only affected files should mention it again I guess?

File: lib/less/importsPush.js (hunk 4 above)
Comment (matz3) [reply lib/less/importsPush.js]: Actually there is not such a statement in the original code. As it's just an import to make the original code work I would expect that this is not relevant.

File: lib/less/fileLoader.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (matz3) [reply lib/less/fileLoader.js]: The link was just a copy-paste error. The file is now unchanged.

**Pull request merged**